rubi/rubi/client.py

```diff
@@ -766,3 +789,49 @@ def _update_asset_allowance(
     @staticmethod
     def _check_allowance(pair: Pair, order: BaseNewOrder):
         pass
+
+    def _handle_transaction_receipt_raw_events(
+        self, transaction_receipt: TransactionReceipt
+    ) -> TransactionReceipt:
+        """
+        Transforms the raw transaction receipt events to human-readable events
+
+        :param transaction_receipt:
+        :type transaction_receipt: TransactionReceipt
+        :return: The transaction receipt with human-readable events populated
+        :rtype: TransactionReceipt
+        """
+        events: List[OrderEvent] = []
+        for raw_event in transaction_receipt.raw_events:
+            if (
+                isinstance(raw_event, EmitFeeEvent)
+                or isinstance(raw_event, EmitSwap)
+                or isinstance(raw_event, Dict)
+            ):
+                # TODO: handle these events correctly
+                continue
+            else:
+                raw_event: Union[EmitOfferEvent, EmitTakeEvent, EmitCancelEvent]
+
+                event_pair = None
+                for pair in self._pairs.values():
+                    if (
+                        pair.ask_identifier == raw_event.pair
+                        or pair.bid_identifier == raw_event.pair
+                    ):
+                        event_pair = pair
+                        break
+
+                if not event_pair:
```

Comment:
in theory, we could handle the `pair` field by performing the same `keccak256(abi.encodePacked(_offer.pay_gem, _offer.buy_gem)` that is occurring when the event is created to the assets in the event. since this is implemented at the client level, i think it is fine that the user must have this `pair` registered, but if we move away from this `pair` based approach, this could be an approach we take